From 27bfa5a64933a46cb3b6edc73d64fd401953ea09 Mon Sep 17 00:00:00 2001
From: bendikro <bendikro@gmail.com>
Date: Tue, 21 May 2013 18:36:21 +0200
Subject: [PATCH 4/5] Fix #2302 : deluge-gtk crashed with IndexError in
 get_plugin_info

This fix handles the plugin info not being available
---
 deluge/pluginmanagerbase.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/deluge/pluginmanagerbase.py b/deluge/pluginmanagerbase.py
index da46449..c34b9d3 100644
--- a/deluge/pluginmanagerbase.py
+++ b/deluge/pluginmanagerbase.py
@@ -174,6 +174,12 @@ def get_plugin_info(self, name):
         info = {}.fromkeys(METADATA_KEYS)
         last_header = ""
         cont_lines = []
+        # Missing plugin info
+        if not self.pkg_env[name]:
+            log.warn("Failed to retrive info for plugin '%s'" % name)
+            for k in info:
+                info[k] = _("Not available")
+            return info
         for line in self.pkg_env[name][0].get_metadata("PKG-INFO").splitlines():
             if not line:
                 continue
@@ -187,5 +193,4 @@ def get_plugin_info(self, name):
                 if line.split(":", 1)[0] in info.keys():
                     last_header = line.split(":", 1)[0]
                     info[last_header] = line.split(":", 1)[1].strip()
-
         return info
-- 
1.8.3.1

